draft-ietf-mls-protocol.md

```diff
@@ -954,25 +981,24 @@ Given these inputs, the derivation of secrets for an epoch
 proceeds as shown in the following diagram:
 
 ~~~~~
-               Init Secret [n-1] (or 0)
+               init_secret [n-1] (or 0)
                      |
                      V
-Update Secret -> HKDF-Extract = Epoch Secret
-                     |
-                     |
-                     +--> Derive-Secret(., "msg", ID, Epoch, Msg)
-                     |       = message_master_secret
+update_secret -> HKDF-Extract = epoch_secret
                      |
                      +--> Derive-Secret(., "add", ID, Epoch, Msg)
                      |       |
                      |       V
-                     |    Derive-Key-Pair(.) = Add Key Pair
+                     |    Derive-Key-Pair(.) = add_key_pair
+                     |
+                     +--> Derive-Secret(., "app", ID, Epoch, Msg)
+                     |    = application_secret_[0]
                      |
                      V
                Derive-Secret(., "init", ID, Epoch, Msg)
                      |
                      V
-               Init Secret [n]
+               init_secret [n]
```

Comment:
If you're going to use `application_secret_[0]`, this should be `init_secret_[n]`.

Reply:
Fixed, Thanks !